### PR TITLE
Use PimcoreLocationAwareConfigDao

### PR DESCRIPTION
**Pull request by blankse**
Add possibility to save the qr codes in yaml or settings-store db table

### CODE REVIEW

**Comment by blankse** [comment]
@GALCF Can you review this?

File: src/DependencyInjection/PimcoreQrcodeExtension.php [hunk 1]
```diff
@@ -2,25 +2,55 @@
...
 use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\HttpKernel\DependencyInjection\ConfigurableExtension;
 
-class PimcoreQrcodeExtension extends Extension
+class PimcoreQrcodeExtension extends ConfigurableExtension implements PrependExtensionInterface
 {
     /**
-     * @param array<mixed> $configs
+     * @param array $mergedConfig
```
Comment (GALCF) [inline src/DependencyInjection/PimcoreQrcodeExtension.php]:
```suggestion
     * @param array<mixed> $mergedConfig
```

Comment (GALCF) [reply src/DependencyInjection/PimcoreQrcodeExtension.php]: You can run `php vendor/bin/phpstan analyze src --level=8 --memory-limit=-1` to check for static analysis issues

File: src/DependencyInjection/PimcoreQrcodeExtension.php [hunk 2]
```diff
@@ -2,25 +2,55 @@
...
+     */
+    public function prepend(ContainerBuilder $container)
+    {
+        if ($container->hasExtension('doctrine_migrations')) {
+            $loader = new YamlFileLoader(
+                $container,
+                new FileLocator(__DIR__ . '/../Resources/config')
+            );
+
+            $loader->load('doctrine_migrations.yml');
```
Comment (GALCF) [inline src/DependencyInjection/PimcoreQrcodeExtension.php]: Could you change the file extension to `.yaml` please

**Comment by GALCF** [comment]
> @GALCF Can you review this?

Looks great, thank you so much!  
I'll add the comments via suggestions in a bit

**Review by GALCF — CHANGES_REQUESTED** [review]

**Pull request merged**